lib/spack/spack/test/relocate.py

```diff
@@ -279,38 +300,72 @@ def test_relocate_elf_binaries_absolute_paths(hello_world, tmpdir):
 
 
 @pytest.mark.requires_executables('patchelf', 'strings', 'file', 'gcc')
-def test_relocate_elf_binaries_relative_paths(hello_world, tmpdir):
+def test_relocate_elf_binaries_relative_paths(hello_world, copy_binary):
     # Create an executable, set some RPATHs, copy it to another location
-    orig_binary = hello_world(
-        rpaths=['$ORIGIN/lib', '$ORIGIN/lib64', '/opt/local/lib']
-    )
-    new_root = tmpdir.mkdir('another_dir')
-    shutil.copy(str(orig_binary), str(new_root))
+    orig_binary = hello_world(rpaths=['lib', 'lib64', '/opt/local/lib'])
+    new_binary = copy_binary(orig_binary)
 
-    # Relocate the binary
-    new_binary = new_root.join('main.x')
     spack.relocate.relocate_elf_binaries(
         binaries=[str(new_binary)],
         orig_root=str(orig_binary.dirpath()),
-        new_root=str(new_root),
-        new_prefixes={str(tmpdir): '/foo'},
+        new_root=str(new_binary.dirpath()),
+        new_prefixes={str(orig_binary.dirpath()): '/foo'},
         rel=True,
         orig_prefix=str(orig_binary.dirpath()),
-        new_prefix=str(new_root)
+        new_prefix=str(new_binary.dirpath())
     )
 
-    # Check that the RPATHs changed
-    patchelf = spack.util.executable.which('patchelf')
-    output = patchelf('--print-rpath', str(new_binary), output=str)
-
     # Some compilers add rpaths so ensure changes included in final result
-    assert '/foo/lib:/foo/lib64:/opt/local/lib' in output
+    assert '/foo/lib:/foo/lib64:/opt/local/lib' in rpaths_for(new_binary)
+
+
+@pytest.mark.requires_executables('patchelf', 'strings', 'file', 'gcc')
+def test_make_elf_binaries_relative(hello_world, copy_binary, tmpdir):
+    orig_binary = hello_world(rpaths=[
+        str(tmpdir.mkdir('lib')), str(tmpdir.mkdir('lib64')), '/opt/local/lib'
+    ])
+    new_binary = copy_binary(orig_binary)
+
+    spack.relocate.make_elf_binaries_relative(
+        [str(new_binary)], [str(orig_binary)], str(orig_binary.dirpath())
+    )
+
+    assert rpaths_for(new_binary) == '$ORIGIN/lib:$ORIGIN/lib64:/opt/local/lib'
+
+
+def test_raise_if_not_relocatable(monkeypatch):
+    monkeypatch.setattr(spack.relocate, 'file_is_relocatable', lambda x: False)
+    with pytest.raises(spack.relocate.InstallRootStringError):
+        spack.relocate.raise_if_not_relocatable(
+            ['an_executable'], allow_root=False
+        )
+
+
+@pytest.mark.requires_executables('patchelf', 'strings', 'file', 'gcc')
+def test_relocate_text_bin(hello_world, copy_binary, tmpdir):
+    orig_binary = hello_world(rpaths=[
+        str(tmpdir.mkdir('lib')), str(tmpdir.mkdir('lib64')), '/opt/local/lib'
+    ])
+    new_binary = copy_binary(orig_binary)
+
+    # Check this call succeed
+    spack.relocate.relocate_text_bin(
+        [str(new_binary)],
+        str(orig_binary.dirpath()), str(new_binary.dirpath()),
+        spack.paths.spack_root, spack.paths.spack_root,
+        {str(orig_binary.dirpath()): str(new_binary.dirpath())}
+    )
```

Comment:
Shouldn't we actually test that the replacement succeeded? We could read it in as binary ('rb' mode) and check that it contains the new string and not the old string.

Reply:
Nice idea. I probably have to modify the `hello_world` fixture to contain one of these paths in the data segment, but I think this is absolutely doable.

Reply:
Done in b9de3b2